### ISSUE TITLE
Time minute interval does not roll hours correctly

### ISSUE THREAD

**Issue by ram-one** [mbostock/d3#2522]
If I use a minute range of 45 minutes, the generated range is not valid.
```
var start = new Date(1439485200000),
     end =  d3.time.hour.offset( new Date(1439485200000), 3);
d3.time.minute.range(start, end, 45);
```
produces 17:00 - 17:45 - 18:00 - 18:45 - 19:00 - 19:45.
I was expecting that it would be 
17:00 - 17:45 - 18:30 - 19:15

It happens with every step value that is not a divider of 60.

**Comment by mbostock** [comment]
Yes. The bug is more general, in that for any time interval, interval.range assumes that the maximum unit of time is divisible by the step. So for minutes, the maximum is 60, and the code only works correctly for steps such as 1, 2, 5, 10, 15, 30. Since 60 is not an integer multiple of 45, the wrong result is returned.

FWIW, this is fixed in [d3-time](https://github.com/d3/d3-time) for 4.0:

```
> time = require("d3-time")
…
> start = new Date(1439485200000)
Thu Aug 13 2015 10:00:00 GMT-0700 (PDT)
> end = time.hour.offset(start, 3)
Thu Aug 13 2015 13:00:00 GMT-0700 (PDT)
> time.minute.range(start, end, 45)
[ Thu Aug 13 2015 10:00:00 GMT-0700 (PDT),
  Thu Aug 13 2015 10:45:00 GMT-0700 (PDT),
  Thu Aug 13 2015 11:30:00 GMT-0700 (PDT),
  Thu Aug 13 2015 12:15:00 GMT-0700 (PDT) ]
```

**Comment by supriyantomaftuh** [comment]
Yes. The bug is more general, in that for any time interval, interval.range assumes that the maximum unit of time is divisible by the step. So for minutes, the maximum is 60, and the code only works correctly for steps such as 1, 2, 5, 10, 15, 30. Since 60 is not an integer multiple of 45, the wrong result is returned.

FWIW, this is fixed in d3-time for 4.0:

> time = require("d3-time")
…
> start = new Date(1439485200000)
Thu Aug 13 2015 10:00:00 GMT-0700 (PDT)
> end = time.hour.offset(start, 3)
Thu Aug 13 2015 13:00:00 GMT-0700 (PDT)
> time.minute.range(start, end, 45)
[ Thu Aug 13 2015 10:00:00 GMT-0700 (PDT),
  Thu Aug 13 2015 10:45:00 GMT-0700 (PDT),
  Thu Aug 13 2015 11:30:00 GMT-0700 (PDT),
  Thu Aug 13 2015 12:15:00 GMT-0700 (PDT) ]

**Comment by mbostock** [comment]
Closing as this is fixed in d3-time for 4.0.
